java/fury-core/src/main/java/org/apache/fury/resolver/ClassResolver.java

```diff
@@ -728,13 +731,17 @@ private void addSerializer(Class<?> type, Serializer<?> serializer) {
       }
       classInfo = classInfoMap.get(type);
     }
+
     if (classInfo == null || typeTag != null || classId != classInfo.classId) {
-      classInfo = new ClassInfo(this, type, typeTag, serializer, classId);
-    } else {
-      classInfo.serializer = serializer;
+      classInfo = new ClassInfo(this, type, typeTag, null, classId);
+      classInfoMap.put(type, classInfo);
+      if (registered) {
+        registeredId2ClassInfo[classId] = classInfo;
+      }
     }
-    // make `extRegistry.registeredClassIdMap` and `classInfoMap` share same classInfo instances.
```

Comment:
Here it should be that `classInfoMap` and `registeredId2ClassInfo` share the same ClassInfo.

The ClassInfo in `registeredId2ClassInfo` and `classInfoMap` should always be shared. The `ClassInfo` obtained in `registeredId2ClassInfo` can be the same in `classInfoMap`, so there should be redundancy here, only whether to create ClassInfo to `classInfoMap `Just put it in